Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp

```diff
@@ -505,9 +505,9 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<ClassicScript>> fetch_a_classic_worker_impo
 
     // 5. Pause until response is not null.
     auto& event_loop = settings_object.responsible_event_loop();
-    event_loop.spin_until([&]() {
+    event_loop.spin_until(JS::create_heap_function(vm.heap(), [&]() -> bool {
```

Comment:
alternatively i was thinking event loop could create the heap function itself and this just takes a standard `Function` since this is definitely a bit uglier, but eh, dont know